Run go-fusefpt via gRPC so that we don't require a fuse installation

thallgren opened the pull request:
If we link the telepresence binary with a fuse library, then that
library must be installed in order for telepresence to run. This is a
problem on platforms like macOS and Windows where the fuse installation
is either proprietary or GPL3, because we cannot bundle it with our
installation.

Running a linked binary via gRPC will not stop the telepresence binary
from functioning in case the gRPC server doesn't start.

Commit:
Revert "Fix cross-compilation issue when building arm64 on linux"

This reverts commit a7640f9a37b8694b30bcb805ac9bf7a7922934b9.

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

Commit:
Revert "Ensure that fuse is installed prior to building release"

This reverts commit 29dd6ee25f6dd4a8dc06609b05be197924b89826.

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

Commit:
Run go-fusefpt via gRPC so that we don't require a fuse installation

If we link the telepresence binary with a fuse library, then that
library must be installed in order for telepresence to run. This is a
problem on platforms like macOS and Windows where the fuse installation
is either proprietary or GPL3, because we cannot bundle it with our
installation.

Running a linked binary via gRPC will not stop the telepresence binary
from functioning in case the gRPC server doesn't start.

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

josecv commented on pkg/client/userd/fuseftp.go:
```diff
@@ -0,0 +1,123 @@
+	}
+
+	downloadURL := fmt.Sprintf(latestFuseFTPRelease, runtime.GOOS, runtime.GOARCH, suffix)
+	dlog.Debugf(ctx, "About to download telepresence-pro from %s", downloadURL)
```
this should say `go-fuseftp`

josecv reviewed (approved):
nit re logging

josecv committed:
fix a log line

Signed-off-by: Jose Cortes <josecortes@datawire.io>

josecv committed:
Bump go to 1.19 in license check

Signed-off-by: Jose Cortes <josecortes@datawire.io>

Commit:
Make the grpc.Dial work on all platforms

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

Commit:
Add cross-compiler so that we can build CGO arm64 binaries on Linux

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

Commit:
Embed the bits for the fuseftp binary into telepresence

This removes the need for a public network connection when using the
fuseftp solution with Telepresence.

Signed-off-by: Thomas Hallgren <thomas@datawire.io>
